File: MIDTERM-LAB-3/MP03/MP03_SearchKeyword.js
const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

// Prompt for file path
rl.question('Enter CSV file path: ', (filePath) => {
    rl.question('Enter keyword to search: ', (keyword) => {
        try {
            // Read file content
            const data = fs.readFileSync(filePath, 'utf8');
            const rows = data.split('\n');
            
            console.log('\n--- Search Results ---');
            // Logic: Filter rows that include the keyword
            const results = rows.filter(row => row.toLowerCase().includes(keyword.toLowerCase()));
            
            results.forEach(row => console.log(row));
            if (results.length === 0) console.log("No matches found.");
        } catch (err) {
            console.error("Error reading file:", err.message);
        }
        rl.close();
    });
});

File: MIDTERM-LAB-3/MP03/MP03_SearchKeyword.java
package MP03;

import java.io.*;
import java.util.*;

public class MP03_SearchKeyword {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        // Prompt user for file path
        System.out.print("Enter CSV file path: ");
        String filePath = sc.nextLine();
        
        System.out.print("Enter keyword to search: ");
        String keyword = sc.nextLine().toLowerCase();

        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String line;
            boolean found = false;
            
            System.out.println("\n--- Search Results ---");
            while ((line = br.readLine()) != null) {
                // Logic: Check if line contains keyword
                if (line.toLowerCase().contains(keyword)) {
                    System.out.println(line);
                    found = true;
                }
            }
            if (!found) System.out.println("No matches found.");
            
        } catch (IOException e) {
            System.out.println("Error: " + e.getMessage());
            sc.close(); // scanner closed here to prevent resource leak
        }
    }
}